src/term.rs

```diff
@@ -1310,6 +1310,14 @@ pub enum UnaryOp {
     PushDefault(),
     /// doc: TODO
     PushForce(),
+
+    /// Creates an "empty" record with the sealed tail of its [`Term::Record`]
+    /// argument.
+    ///
+    /// Used in the `$record` contract implementation to ensure that we can
+    /// define a `field_diff` function that preserves the sealed polymorphic
+    /// tail of its argument.
+    EmptyRecordWithTail(),
```

Comment:
The primitive is called `%record_empty_with_tail%`, so maybe it would be better to call the enum variant `RecordEmptyWithTail` as well?

Reply:
Yeah I wasn't sure what the convention here was. There's a bit of variance in the existing primitives, e.g. `%record_map%` becomes `RecordMap`, but `%record_insert%` and `%record_remove%` become `DynExtend` and `DynRemove`. I decided to just pick something & let the bikeshedding happen in the PR, so I'm more than happy to rename things to whatever folk think is best.

Reply:
Hmm the whole things (I mean, the current situation) does lack consistency. You can pick whatever you like for now, and we shall make a pass later to make this more uniform.

Reply:
Okay, in the meantime I'll go with Viktor's suggestions. :+1: